src/main/java/spleetwaise/transaction/model/transaction/Amount.java

```diff
@@ -0,0 +1,78 @@
+package spleetwaise.transaction.model.transaction;
+
+import static java.util.Objects.requireNonNull;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+import spleetwaise.address.commons.util.AppUtil;
+
+/**
+ * Represents a Transaction's amount in the transaction book.
+ * Guarantees: immutable; is valid or declared in {@link #isValidAmount(String)}
+ */
+public class Amount {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Amount should only contain digits up to 2 decimal points delimited by . and prefixed with +/-";
+
+    /*
+     * The first character of amount must be + or - and only allow precision up to 2 decimal places
+     */
+    public static final String VALIDATION_REGEX = "^(\\+|\\-)([\\d]+$|[\\d]+\\.[\\d]{1,2}$)";
```

Comment:
Not sure if this regex is correct. @seanlim Could you help to check?
Maybe `^[+-]?\d+(\.\d{1,2})?$`?